### ISSUE TITLE
Google Drive problems

### ISSUE THREAD

**Issue by Ugo-Gira** [Cimbali/CleanLinks#133]
Hi, when opening a link to a video in Google Drive, the black page with the video thumbnail loads, but when I click on the play button the video loads forever and never plays. I cannot whitelist because according to CleanLinks no link has been cleaned.
But as soon as I disable the Addon, I can watch the video.
Thanks

**Comment by Cimbali** [comment]
Sorry about that, I will try to investigate but I'm not sure what's going on there.

**Comment by Ugo-Gira** [comment]
It might be something you do wrong with Google.
I notice now that when I use Google Shopping, when I click "to the website"
in one of the results in the list of all the idea for any given article, it
tries to redirect and stays as a white page forever.

If you tell me how I can manually whitelist my typing the domain somewhere
instead of having to select a link in your add-on (which in these two
Google cases it's impossible as no link is shown) I can workaround this way
while you fix the issue, otherwise I need to uninstall.

Thanks
Ugo

**Comment by Cimbali** [comment]
I don't think we do anything specific with Google, but to whitelist it you can create a new rule on the rules editing page (accessible eg via the options) with:

- domain `google.*`
- check matching sub domains 
- whitelist parameter `.*`
- check allow javascript to modify links 

Then click the save rule button.

**Comment by Cimbali** [comment]
It might be that something in the url looks like an embedded url when it's not a valid url? An example page of Google shopping or Google drive where this happens would be most helpful.

**Comment by Cimbali** [comment]
I found a random drive with some game demo: https://drive.google.com/drive/folders/1GSR0ThesQIYsPY2wn3qaigCpRzYjVn7k

Does this reproduce your issue ?

I managed to fix it with the following rules:
- domain `docs.google.com/` path `^/get_video_info$` whitelist parameters `eurl`
- domain `docs.google.com/` path `^/videoplayback$` whitelist `ei` and `source`
- domain `youtube.googleapis.com` path `^/embed/$` remove parameters `BASE_URL`, `BASE_YT_URL`, `iurl`, `reportabuseurl`
